File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/request/ChangeUsernameRequest.java
package dev.starryeye.book_library.controller.user.request;

import dev.starryeye.book_library.application.command.command.ChangeUsernameCommand;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record ChangeUsernameRequest(
        @NotNull(message = "id must not be null")
        Long id,
        @NotBlank(message = "username must not be blank")
        String username
) {

    public ChangeUsernameCommand toCommand() {
        return new ChangeUsernameCommand(id, username);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/request/ChangeUsernameRequest.kt
package dev.starryeye.book_library.controller.user.request

import dev.starryeye.book_library.application.user.command.command.ChangeUsernameCommand
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class ChangeUsernameRequest(
    @field:NotNull
    val id: Long?,
    @field:NotBlank
    val username: String?
)

fun ChangeUsernameRequest.toCommand(): ChangeUsernameCommand {
    return ChangeUsernameCommand(
        id = id!!,
        username = this.username!!,
    )
}